### DISCUSSION TITLE
Combo box with empty option to allow submitting without double clicking enter key

### DISCUSSION THREAD

**Discussion by RaenonX** [tailwindlabs/headlessui#3382] [Ideas]
https://codesandbox.io/p/sandbox/tailwind-combobox-r48pw3?file=%2Fsrc%2FApp.tsx%3A23%2C50

For the combo box allowing custom value and no options candidate, editing the value then click on enter should just submit the form.

## Reproduction?

1. Open console
2. Type whatever in the input box
3. Hit "enter"
4. Notice that the first hit doesn't submit the form, it needs 2 hits.

I think this is because that whenever the input value has changed, regardless if it actually has options or not, it marks the combobox as "opened." So the 1st hit closes it, then the 2nd hit actual submits it.

## Expected / Wanted Behavior

Without any options, simply hitting enter should just submit the form. If empty candidate is something possibly to worry about (so to mark the combobox as open), maybe add some `boolean` attribute that dictates the behavior on empty?

----------------

As always, thanks for creating this lib.

**Comment by kb1995** [comment]
Hey @RaenonX did you find a solution for this?

**Reply by RaenonX** [reply]
Unfortunately no, ended up just not using this implementation :/
